h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamUtilsThinPlateRegression.java

```diff
@@ -0,0 +1,53 @@
+package hex.genmodel.algos.gam;
+
+public class GamUtilsThinPlateRegression {
+  public static double calTPConstantTerm(int m, int d, boolean dEven) {
+    if (dEven)
+      return (Math.pow(-1, m + 1 + d / 2.0) / (Math.pow(2, 2*m - 1) * Math.pow(Math.PI, d / 2.0) * 
+              factorial(m-1)*factorial(m - d / 2)));
```

Comment:
is this supposed to calculate terms before r_i from 3.1? If yes, why there is factorial(m-1) term?

Reply:
Zuzana:

Thank you for checking it.  I forgot to upload the latest formula to the JIRA.  Please download the new document from the JIRA.  The correct formula is 
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/7231712/111013841-de4d5180-8355-11eb-8eb9-5c4ffacf4451.png">

Reply:
okok thanks!